prometheus-pushgateway.yaml

```
@@ -33,13 +23,20 @@ pipeline:
         golang.org/x/oauth2@v0.27.0
         golang.org/x/net@v0.38.0
 
-  - runs: |
-      make build
+  - uses: go/build
+    with:
+      packages: .
+      output: prometheus-pushgateway
+      ldflags: |
+        -X github.com/prometheus/common/version.Version=${{package.version}}
+        -X github.com/prometheus/common/version.Revision=$(git rev-parse --short HEAD)
+        -X github.com/prometheus/common/version.Branch=HEAD
+        -X github.com/prometheus/common/version.BuildUser=reproducible@reproducible
+        -X github.com/prometheus/common/version.BuildDate=$(date -u -d "@$SOURCE_DATE_EPOCH" +%Y-%m-%d-%H:%M)
 
+  # for backward compatibility with older installations expecting pushgateway in /usr/bin
   - runs: |
-      install -Dm755 pushgateway "${{targets.destdir}}"/usr/bin/pushgateway
-
-  - uses: strip
+      ln -sf prometheus-pushgateway ${{targets.destdir}}/usr/bin/pushgateway
```

```suggestion
      ln -sf /usr/bin/prometheus-pushgateway ${{targets.destdir}}/usr/bin/pushgateway
```

And also, can we have a `readlink` test below? Like this:

```
test "$(readlink /usr/bin/pushgateway)" = "/usr/bin/prometheus-pushgateway"
```

Then

```
pushgateway --help
```